UnityProject/Assets/Scripts/Objects/Machines/ServerMachines/Communications/CommsServer.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using Communications;
+using InGameEvents;
+using Managers;
+using Systems.Electricity;
+using Systems.Explosions;
+using UnityEngine;
+
+namespace Objects.Machines.ServerMachines.Communications
+{
+	public class CommsServer : SignalReceiver, IEmpAble
+	{
+		[SerializeField] private BlackboxMachine blackbox;
+		[SerializeField] private APCPoweredDevice apcPoweredDevice;
+		private Integrity integrity;
+		private bool isMalf = false;
```

Comment:
Maybe it would be better to spell out "Malfunctioning"? First I thought of malf AI because that's where I heard it used. Someone not familiar with SS13 will probably have trouble understanding it too.